pg-index-health/src/main/java/io/github/mfvanek/pg/checks/host/CheckNotValidConstraintsOnHost.java

```diff
@@ -0,0 +1,30 @@
+package io.github.mfvanek.pg.checks.host;
+
+import io.github.mfvanek.pg.common.maintenance.Diagnostic;
+import io.github.mfvanek.pg.connection.PgConnection;
+import io.github.mfvanek.pg.model.PgContext;
+import io.github.mfvanek.pg.model.column.Column;
+import io.github.mfvanek.pg.model.constraint.ForeignKey;
+import io.github.mfvanek.pg.utils.ColumnsInForeignKeyParser;
+
+import java.util.List;
+import javax.annotation.Nonnull;
+
+public class CheckNotValidConstraintsOnHost extends AbstractCheckOnHost<ForeignKey> {
+
+    public CheckNotValidConstraintsOnHost(@Nonnull final PgConnection pgConnection) {
+        super(ForeignKey.class, pgConnection, Diagnostic.CHECK_NOT_VALID_CONSTRAINTS);
```

Comment:
Please take a look at my comment https://github.com/mfvanek/pg-index-health/issues/362#issuecomment-2044247293

I think we need a new `Constraint `class in model

Reply:
```java
package io.github.mfvanek.pg.model.constraint;

import javax.annotation.Nonnull;
import java.util.Objects;

public enum ConstraintType {

    CHECK("c"),
    FOREIGN_KEY("f");

    // See https://www.postgresql.org/docs/current/catalog-pg-constraint.html
    private final String pgConType;

    ConstraintType(@Nonnull final String pgConType) {
        this.pgConType = Objects.requireNonNull(pgConType, "pgConType");
    }
}
```

and
```java
package io.github.mfvanek.pg.model.constraint;

import io.github.mfvanek.pg.model.DbObject;
import io.github.mfvanek.pg.model.table.TableNameAware;

import javax.annotation.concurrent.Immutable;

@Immutable
public class Constraint implements DbObject, TableNameAware {

    private final String tableName;
    private final String constraintName;
    private final ConstraintType constraintType;
    
    ...
}
```

Reply:
For example see `IndexWithColumns` in https://github.com/mfvanek/pg-index-health/pull/377